bears/jinja2/Jinja2Bear.py

```diff
@@ -0,0 +1,271 @@
+                            suggested.
+        * Control spacing:  Like variable spacing, but for control blocks,
+                            i.e. if and for constructs. Looks at both start
+                            and end block.
+        * Control labels:   It is good practise to label the end of an if or
+                            for construct with a comment equal to the content
+                            of the start, like so:
+
+                            ```
+                            {% for x in y %}
+                            do something
+                            {% endfor %}{# for x in y %}
+                            ```
+
+                            Mising or differing labels are detected and
+                            fixes suggested.
+                            Constructs with start and end on the same line are
+                            being ignored.
+        * unbalanced blocks:   Each opening tag for a for or if construct must
+                               be closed by a corresponding end tag. An
+                               unbalanced number of opening and closing tags is
+                               invalid syntax and will be reported with MAJOR
+                               severity by the bear.
+
+
+        :param variable_spacing: The number of spaces a variable block
+                                 should be spaced with. Default is 1.
+        :param control_spacing:  The number of spaces a control block
+                                 should be spaced with. Default is 1.
+        '''
+        variable_regex = re.compile(
+            r"(?P<open>{{)(?P<content>.*?)(?P<close>}})")
+        control_start_regex = re.compile(
+            r"(?P<open>{%)(?P<content>\s*(for|if).*?)(?P<close>%})")
+        control_end_regex = re.compile(
+            r"(?P<open>{%)(?P<content>\s*end(for|if)\s*?)"
+            r"(?P<close>%})(?P<label>{#.*?#})?")
+
+        # Whenever a control construct starts a tuple of the
+        # expected end label and line number are put on this stack.
+        # Whenever an end tag is encountert the item on top is popped.
+        control_stack = []
+
+        for line_number, line in enumerate(file, start=1):
+
+            # check for variable spacing issues
+            for m in variable_regex.finditer(line):
+                match = m.group("content")
+                if not is_spacing_valid(match, variable_spacing):
+                    diff = generate_spacing_diff(
+                        file, filename, line, line_number, m, variable_spacing)
+                    yield Result.from_values(
+                        origin=self,
+                        message="Variable blocks should be spaced with "
+                                "`{0}` spaces on each side.".format(
+                                    variable_spacing),
+                        file=filename,
+                        line=line_number,
+                        column=m.start(0) + 1,
```

Comment:
Ah, I see. Could you comment `0 => the whole match` or so, was a bit confused in the start (Probably because of my lack of understanding on re though... But a comment doesn't hurt)